src/view/com/util/UserAvatar.tsx

```diff
@@ -159,16 +169,18 @@ export {DefaultAvatar}
 
 let UserAvatar = ({
   type = 'user',
+  shape,
   size,
   avatar,
   moderation,
   usePlainRNImage = false,
 }: UserAvatarProps): React.ReactNode => {
   const pal = usePalette('default')
   const backgroundColor = pal.colors.backgroundLight
+  shape = shape || type === 'user' ? 'circle' : 'square'
```

Comment:
I struggle to read this — is the precedence correct? I think the intent is

```js
shape = shape || (type === 'user' ? 'circle' : 'square')
```

with the part after `||` being a default value.

But the code is doing something more like `'foo' || true ? 1 : 0` (which gives `1` rather than `'foo'`)

Overall having two sources of truth for this is confusing, is the intent to allow sometimes specifying a non-type-default shape?

Reply:
ah whoops

```
shape = shape || (type === 'user' ? 'circle' : 'square')
```

Correct that ^ is the intent. Essentially `shape` acts as an override of the shape that the different types default to. In this case, labelers are normally square, but when theyre on posts they need to be circles